### PR TITLE
Make builder a class

### PR DESCRIPTION
**Pull request by tryoxiss**
khaim pls review :3

### CODE REVIEW

File: modules/api.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	def modified_datetime():
+		pass
+
+def confirm_output_exists(path):
+	if os.path.exists(get_output_variant(path)) == True:
+		return
+
+	os.mkdir(get_output_variant(path))
+
+# def get_output_variant(path: pathlib.Path) -> str:
```
Comment (khaim0919) [inline modules/api.py:28]: remove unused comment

File: modules/api.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	def __init__(
+		url: str,
+	):
+		pass
+		# give tis all the data that is accessed via the api
+
+	def modified_datetime():
+		pass
+
+def confirm_output_exists(path):
```
Comment (khaim0919) [inline modules/api.py:22]: document function

File: modules/api.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+# 	"""
+# 	Takes a path to a file and replaces the first case of the input directory
+# 	with the output directory to create the output files path.
+# 	"""
+# 	return str(path).replace(config.content.directory, config.output.directory, 1)
+
+# def strip_file(path: str) -> str:
+#	a
+
+def get_output_variant(path: str) -> str:
```
Comment (khaim0919) [inline modules/api.py:38]: this and the comment above seem like the same, could use the previous ones comment as this ones.

File: modules/api.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	#
+	# we also want to allow for pattersn like /posts/12/sep/2023/ and allow writing to multiple locations.
+
+	if path.endswith("index.html"):
+		path = path.rstrip("index.html")
+
+	return path
+
+
+def create_output_directory():
```
Comment (khaim0919) [inline modules/api.py:64]: want doc comment here but I wrote this function so Ill do that after you commit this

File: modules/builder.py [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+	def full_name(self):
+		pass
+
+
+	def path_only(self):
+		return self.path.rstrip(self.name())
+
+
+	def make_path_fancy(self, path: str) -> str:
```
Comment (khaim0919) [inline modules/builder.py:52]: duplicate function name in modules/api.py

File: modules/builder.py [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+
+
+		# This bit is a bit ew but basically if its something like
+		# */name1/name1.* then we just strip the end bit and replace
+		# it with index.html to make it work the way we want it to.
+
+		# example:
+		# posts/post-name/post-name.md
+		if path.endswith(f"{self.name().rstrip(self.extension())}/{self.name()}"):
+			print("VHAD!!")
```
Comment (khaim0919) [inline modules/builder.py:85]: reeemove print?

**Review by khaim0919 — COMMENTED** [review]
only some little things, ill start soon as you commit this !

File: modules/builder.py (hunk 5 above)
Comment (tryoxiss) [reply modules/builder.py:52]: it only exists once so its not duplicated.

File: modules/api.py (hunk 4 above)
Comment (tryoxiss) [reply modules/api.py:64]: This is self explanatory, no doc comment needed.

File: modules/builder.py (hunk 5 above)
Comment (tryoxiss) [reply modules/builder.py:52]: Oh oops nvm, fixed

**Pull request merged**